### DISCUSSION TITLE
A quick question or two

### DISCUSSION THREAD

**Discussion by rlgordey** [rabbitmq/rabbitmq-server#3721] [Questions] [answered]
When I install RabbitMQ on a Windows Box it sets the config and data directory to `%APPDATA%\RabbitMQ` which is translated to the user's home directory that is preforming the install. I know on Linux it uses `${install_prefix}/etc/rabbitmq/rabbitmq`. Why on Windows, does the installer not put the config or data directory on something like `C:\ProgramData\RabbitMQ Server`?

Luckily I found this [Web Article](https://www.rpic.com/knowledgebase/imagenow/changing-rabbitmq_base-directory-windows-installation/).

Having said that, when I upgrade said Windows box from RabbitMQ 3.9.8 to 3.9.9 it doesn't just upgrade and keep the existing config. I have to uninstall RabbitMQ completely and again follow the Web article above just to upgrade... Even on a Windows box with the default install I have to completely uninstall and reinstall to get it to work.

I understand that to make RabbitMQ work on various platforms some concessions had to be made, but for Windows this seems overly complicated for no reason.

**Comment by mkuratczyk** [comment]
Personally I probably won't be able to help with a Windows-related issue but my first question would be: what do you mean by "I have to completely uninstall"? What happens if you don't?

**Reply by rlgordey** [reply]
The install won't run as it can't find the config and data files. So I have to uninstall and install fresh.

**Comment by lukebakken** [comment]
> Why on Windows, does the installer not put the config or data directory on something like C:\ProgramData\RabbitMQ Server?

The RabbitMQ team has not had the time to improve the Windows installation experience. At some point a paying customer who uses Windows may provide funds to improve how RabbitMQ works on Windows.

> Having said that, when I upgrade said Windows box from RabbitMQ 3.9.8 to 3.9.9 it doesn't just upgrade and keep the existing config. I have to uninstall RabbitMQ completely and again follow the Web article above just to upgrade

Can you describe *exactly* how you are performing your initial installation on Windows as well as the upgrade? At what point are you setting the `RABBITMQ_BASE` environment variable? How are you setting it, etc etc?

**Reply by rlgordey** [reply]
I run a fresh install but deselect the Service. I set `RABBITMQ_BASE` like this: set RABBITMQ_BASE="C:\ProgramData\RabbitMQ Server" at a RabbitMQ Command Prompt. I then install the RabbitMQ service and run it, just like in the the web article I mentioned.

I get it... you need funds to support a better Windows experience. I just thought I would mention my current difficulties, kinda put a bug in your ear, so to speak. Here's hoping you get a "deep pockets" Sponsor soon.

**Comment by lukebakken** [comment, marked as answer]
I suggest setting `RABBITMQ_BASE` as a **system** environment variable. That way all installers will pick up the value. The value is "baked" into the registry when the Windows service is installed.

I'm reasonably certain that setting the `RABBITMQ_BASE` variable in this manner will enable upgrades to work.

**Reply by rlgordey** [reply]
@lukebakken  This did work upgrading to .version 3.9.10.

**Reply by lukebakken** [reply]
Thank you for letting us know.
